sycl/doc/SYCLInstrumentationUsingXPTI.md

```diff
@@ -0,0 +1,244 @@
+# SYCL Instrumentation
+Any language or programming paradigm must provide mechanisms to correlate a
+developer's use of the language to the debug and performance traces for that
+use. A lightweight tracing framework (XPTI) was developed to enable this for
+SYCL and is the primary mechanism that is employed to enable debug and
+performance traces.
+
+> **NOTE:** For additional information on the XPTI framework, please refer to
+>the [Framework Documentation](https://github.com/intel/llvm/tree/sycl/xptifw/doc/XPTI_Framework.md) for API use
+>and framework performance data.
+
+This document outlines the use of this framework API in the SYCL runtime
+library. The primary concept enable by this framework is the generation of a
+unique 64-bit ID, referred to as the Universal ID (UID), for every public
+language entry point into the library. This allows tools and other helps in the
+software stack to correlate debug and performance data by tagging it with the
+64-bit UID. The framework also provides the ability to propagate this UID all
+the way to the driver layers for the target device so data from lower layers and hardware can be correlated easily.
```

Comment:
Could you please shrink this line to 80 characters?

Reply:
Done!